src/main/java/com/google/sps/authentication/UserAuthenticationStatus.java

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.authentication;
+
+/** Class containing user's authentication status. */
+public class UserAuthenticationStatus {
+
+    private final boolean isLoggedIn;
+
+    // Depends on {@code isLoggedIn}, it will be a login link if logged in, and 
+    // logout link otherwise.
+    private final String url;
```

Comment:
Using the same url variable for both the login- and logout-url may be confusing later. Your intention is clear here, but in the javascript code handling the response of the UserAuthenticationServlet it's ambiguous that you use the same url value for two different cases. 
Having one unused variable is not expensive, so maybe use a separate loginUrl and logoutUrl instead. You can use the builder pattern with optional fields to avoid sending empty strings to the constructor.

Reply:
done